packages/parse5/lib/tokenizer/index.test.ts

```diff
@@ -1,9 +1,51 @@
 import { Tokenizer } from 'parse5';
 import { generateTokenizationTests } from 'parse5-test-utils/utils/generate-tokenization-tests.js';
+import * as assert from 'node:assert';
 
 const dataPath = new URL('../../../../test/data/html5lib-tests/tokenizer', import.meta.url);
 const tokenizerOpts = {
     sourceCodeLocationInfo: true,
 };
 
 generateTokenizationTests('Tokenizer', dataPath.pathname, (handler) => new Tokenizer(tokenizerOpts, handler));
+
+function noop(): void {
+    // Noop
+}
+
+describe('Tokenizer methods', () => {
+    it('should pause and resume', () => {
+        let count = 0;
+        const tokenizer = new Tokenizer(tokenizerOpts, {
+            onComment(t): void {
+                assert.strictEqual(t.data, 'INIT');
+                assert.strictEqual(count++, 0);
+
+                tokenizer.pause();
+                tokenizer.write('<!doctype foo>', false);
+            },
+            onDoctype(t): void {
+                assert.strictEqual(t.name, 'foo');
+                assert.strictEqual(count++, 2);
+
+                expect(() => tokenizer.resume()).toThrow('Parser was already resumed');
+                tokenizer.write('<next>', true);
+            },
+            onStartTag(t): void {
+                assert.strictEqual(count++, 3);
+                assert.strictEqual(t.tagName, 'next');
+            },
+            onEndTag: noop,
```

Comment:
Why are all these required?
Should the default behavior instead become `noop`?

Reply:
The tokenizer callbacks are usually all needed. This test doesn't use a bunch of them as we tightly control the inputs.

Reply:
Your use of “usually” is rather “soft”. Does that mean you see something in making them optional?

Reply:
Sorry for not being clearer. This is the _only_ case where not all callbacks are needed. Every other time, we supply all of the relevant callbacks.